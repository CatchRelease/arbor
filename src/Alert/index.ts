export { default as BannerAlert } from './BannerAlert';
export { default as CardAlert } from './CardAlert';
export { default as InlineAlert } from './InlineAlert';
export { default as PaneAlert } from './PaneAlert';
export { default as ToastAlert } from './ToastAlert';
