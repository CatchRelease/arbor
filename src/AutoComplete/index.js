export { default } from './AutoComplete';
export { default as styledAutoCompleteComponents } from './styledAutoCompleteComponents';
export { default as StyledAutoComplete } from './StyledAutoComplete';
