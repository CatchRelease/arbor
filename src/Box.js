import styled from '@emotion/styled';
import {
  boxShadow,
  color,
  display,
  space,
  width,
  height,
  maxWidth,
  textAlign
} from 'styled-system';

const Box = styled.div`
  box-sizing: border-box;
  ${boxShadow};
  ${color};
  ${display};
  ${height};
  ${maxWidth};
  ${space};
  ${textAlign};
  ${width};
`;

Box.propTypes = {
  ...boxShadow.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...height.propTypes,
  ...maxWidth.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...width.propTypes
};

Box.section = Box.withComponent('section');

export default Box;
