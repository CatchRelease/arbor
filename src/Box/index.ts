export { default, Props as BoxProps } from './Box';
