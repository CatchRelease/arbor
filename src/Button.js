import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { transparentize } from 'polished';
import {
  borderRadius,
  brandFont,
  colors,
  fontWeights,
  lineHeights,
  spacings,
  typography
} from './theme';

const Button = styled.button`
  font-family: ${brandFont};
  font-size: ${typography.text.ui.fontSize};
  border: none;
  color: ${colors.white};
  font-weight: ${fontWeights.medium};
  line-height: ${lineHeights.large};
  outline: none;
  transition: box-shadow 0.3s ease, background 0.2s ease;

  &:enabled {
    cursor: pointer;
  }
`;

Button.CTA = styled(Button.withComponent('button'))`
  background: ${colors.blue};
  border-radius: ${borderRadius.large};
  box-shadow: 0 1px 3px ${transparentize(0.7, colors.blue)},
    inset 0 -1px 0 ${transparentize(0.9, colors.black)};
  padding: ${spacings.small} 0;
  text-transform: uppercase;
  width: 100%;

  &:disabled {
    background: ${colors.grey80};
    box-shadow: none;
  }

  &:hover:enabled {
    background: ${colors.blueLight};
    box-shadow: 0 2px 6px ${transparentize(0.8, colors.blueLight)};
  }

  &:focus:enabled {
    background: ${colors.blue};
    box-shadow: 0 0 0 1px ${colors.white}, ${colors.blueDark} 0 0 0 2px;
  }
`;

Button.CTA.displayName = 'Button.CTA';

Button.CTA.propTypes = {
  disabled: PropTypes.bool
};

Button.CTA.defaultProps = {
  disabled: false
};

export default Button;
