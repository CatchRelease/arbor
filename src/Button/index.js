export { default } from './Button';
export { Variant as ButtonVariant } from './variants';
