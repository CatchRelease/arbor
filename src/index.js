export * from './theme';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as CardActions } from './CardActions';
export { default as CardPreview } from './CardPreview';
export { default as CardRow } from './CardRow';
export { default as Checkbox } from './Checkbox';
export { default as Dropdown } from './Dropdown';
export { default as Flex } from './Flex';
export { default as Grid } from './Grid';
export { default as Heading } from './Heading';
export { default as Icon } from './Icon';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Link } from './Link';
export { default as Menu } from './Menu';
export { default as Pagination } from './Pagination';
export { default as Paragraph } from './Paragraph';
export { default as Popover } from './Popover';
export { default as Text } from './Text';
export { default as Tooltip } from './Tooltip';

export { default as reset } from './reset';
export { default as theme } from './theme';
