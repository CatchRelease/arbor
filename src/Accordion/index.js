import styled from '@emotion/styled';

import Box from '../Box';

const Accordion = styled(Box)`
  border: 1px solid ${props => props.theme.colors.border.default};
  border-radius: ${props => props.theme.borderRadius.large};
`;

export default Accordion;
