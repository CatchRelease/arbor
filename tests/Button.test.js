import React from 'react';
import { create } from 'react-test-renderer';

import { Button, theme } from '../src';

test('renders a CTA button', () => {
  const tree = create(<Button.CTA theme={theme}>CTA</Button.CTA>);

  expect(tree).toMatchSnapshot();
});
