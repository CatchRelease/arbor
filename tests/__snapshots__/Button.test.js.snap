// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Sizes properly renders a jumbo sized button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 3rem;
  font-size: 0.875rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="jumbo"
>
  Button Label
</button>
`;

exports[`<Button /> Sizes properly renders a large sized button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Sizes properly renders a medium sized button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 0.5rem;
  height: 2rem;
  font-size: 0.875rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="medium"
>
  Button Label
</button>
`;

exports[`<Button /> Sizes properly renders a small sized button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 0.5rem;
  height: 1.5rem;
  font-size: 0.75rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="small"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a disabled minimal button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background: none;
  border: 1px solid transparent;
  color: #666666;
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #FAFAFA;
  color: #222222;
}

.emotion-0:active:enabled {
  color: #222222;
  background-color: #EEEEEE;
}

.emotion-0:focus:enabled {
  color: #4466EE;
}

.emotion-0:disabled {
  background: none;
  color: #BBBBBB;
}

<button
  className="emotion-0 emotion-1"
  disabled={true}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a disabled primary button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={true}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a disabled secondary button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background-color: #FFFFFF;
  color: #222222;
  font-weight: 400;
  box-shadow: 0 1px 3px rgba(0,0,0,0.1),0 1px 0 rgba(0,0,0,0.05);
  border: 1px solid #EEEEEE;
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  color: #4466EE;
  box-shadow: 0 1px 7px rgba(0,0,0,0.1);
}

.emotion-0:active:enabled {
  color: #4455CC;
  background-color: #FAFAFA;
  border: none;
}

.emotion-0:focus:enabled {
  outline: #4455CC;
  border: 1px solid #EEEEEE;
  color: #222222;
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={true}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a minimal button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background: none;
  border: 1px solid transparent;
  color: #666666;
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #FAFAFA;
  color: #222222;
}

.emotion-0:active:enabled {
  color: #222222;
  background-color: #EEEEEE;
}

.emotion-0:focus:enabled {
  color: #4466EE;
}

.emotion-0:disabled {
  background: none;
  color: #BBBBBB;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a primary button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background-color: #4466EE;
  color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(68,102,238,0.3),inset 0 -1px 0 rgba(0,0,0,0.1);
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  background-color: #5577FF;
  box-shadow: 0 2px 6px rgba(68,102,238,0.2);
}

.emotion-0:active:enabled {
  background: #4455CC;
}

.emotion-0:focus:enabled {
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="large"
>
  Button Label
</button>
`;

exports[`<Button /> Variants properly renders a secondary button 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  border: 1px solid transparent;
  font-weight: 600;
  line-height: 1.5;
  outline: none;
  -webkit-transition: box-shadow 0.3s ease,background 0.2s ease;
  transition: box-shadow 0.3s ease,background 0.2s ease;
  border-radius: 3px;
  margin-top: 1rem;
  padding: 0 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
  background-color: #FFFFFF;
  color: #222222;
  font-weight: 400;
  box-shadow: 0 1px 3px rgba(0,0,0,0.1),0 1px 0 rgba(0,0,0,0.05);
  border: 1px solid #EEEEEE;
}

.emotion-0:disabled {
  background: #DDDDDD;
  color: #AAAAAA;
  box-shadow: none;
  cursor: not-allowed;
}

.emotion-0:enabled {
  cursor: pointer;
}

.emotion-0:active:enabled {
  box-shadow: none;
}

.emotion-0:focus:enabled {
  border: 1px solid #4455CC;
}

.emotion-0:hover:enabled {
  color: #4466EE;
  box-shadow: 0 1px 7px rgba(0,0,0,0.1);
}

.emotion-0:active:enabled {
  color: #4455CC;
  background-color: #FAFAFA;
  border: none;
}

.emotion-0:focus:enabled {
  outline: #4455CC;
  border: 1px solid #EEEEEE;
  color: #222222;
  box-shadow: 0 0 0 1px #FFFFFF,0 0 0 2px #4455CC;
}

<button
  className="emotion-0 emotion-1"
  disabled={false}
  size="large"
>
  Button Label
</button>
`;
